app/src/main.py

```diff
@@ -14,4 +15,25 @@ def profile():
         age=current_user.age,
         grade=current_user.grade.value,
         logged_in=True
-    )
\ No newline at end of file
+    )
+    
+@main.route('/quiz/<int:quiz_id>', methods=['GET'])
+@login_required
+def quiz_page(quiz_id):
+    user = current_user
+    quiz = Quiz.query.get(quiz_id)
+
+    user_attempted = quiz in user.quizzes
+
+    # Users highest score
+    highest_score = 0
+    if user_attempted:
+        highest_score = quiz.users.filter_by(id=user.id).first().score
```

Comment:
Not sure if you're filtering what you actually want, since this just filters through the list of unique users assigned with a specific quiz and gives you back the current user (a User entry). Using `.score` on it would probably just give you a NoneType object, since the User table doesn't have a `.score` column.